src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -46,6 +46,11 @@ internal abstract class AnalyzerDriver : IDisposable
         private Task _primaryTask;
 
         /// <summary>
+        /// Number of worker tasks processing compilation events and executing analyzer actions.
+        /// </summary>
+        private const int _workerCount = 5;
```

Comment:
Maybe N for an N-core machine (N <= 3), N - 1 for 4 .. 8, N - 2 for N > 8, or some such?